### ISSUE TITLE
Error: There was an error sending your trade offer.  Please try again later. (26)

### ISSUE THREAD

**Issue by Commanche2** [Alex7Kom/node-steam-tradeoffers#104]
I get all the time the error that I describe in the title of this ticket. The sentry file was created on April 30th, and of course there was enough time for SteamGuard to allow trades. I logged into the Steam application and tried to perform a manually trade and was successful. What's wrong? I attach the code below:

```javascript
function makeTrade(tradeUser, urlToken, itemsToTrade) {
	inTrade = true;
	var trimTokenUrl = urlToken.split('&token=');
	var tradeAccesToken = trimTokenUrl[1];
	trade.loadPartnerInventory({
	    partnerSteamId: tradeUser,
	    appId: appid,
	    contextId: contextid
	}, function(err, items) {{
	      trade.makeOffer ({
	            partnerSteamId: tradeUser,
	            accessToken: tradeAccesToken,
	            itemsFromMe: [],
	            itemsFromThem: itemsToTrade,
	      }, function(err, response){
	            if (err) {
	              logger.error(err);
	            }
	      });
	    }
	});
```

I get all the information from database:
- tradeUser is the steamID.
- urlToken is the entire public trade url.
- itemsToTrade is  an array of items which structure is that:
   [ { appid: 730, contextid: 2, amount: 1, assetid: '469444882' } ]

Stack trace:

```
Error: There was an error sending your trade offer.  Please try again later. (26)
    at SteamTradeOffers.<anonymous> (/Users/*****/node_modules/steam-tradeoffers/index.js:427:18)
    at Request.self.callback (/Users/*****/node_modules/steam-tradeoffers/node_modules/request/request.js:368:22)
    at Request.emit (events.js:110:17)
    at Request.<anonymous> (/Users/*****/node_modules/steam-tradeoffers/node_modules/request/request.js:1219:14)
    at Request.emit (events.js:129:20)
    at IncomingMessage.<anonymous> (/Users/*****/node_modules/steam-tradeoffers/node_modules/request/request.js:1167:12)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

I cant find my fault. There's no way to get it run.

P.S.: I have no pending trade offers.

**Comment by scholtzm** [comment]
All of the items in `itemsToTrade` are tradable?

**Comment by Commanche2** [comment]
Ofcourse, it's there since the Steam account was created (more or less). I test with skins that are without any restriction.

**Comment by Commanche2** [comment]
I tried to do manually the trade by coding like this:

```javascript
function makeTrade(tradeUser, urlToken, itemsToTrade) {
	trade.loadPartnerInventory({
	    partnerSteamId: 'Steam id as string',
	    appId: 730,
	    contextId: 2
	}, function(err, items) {{
	      trade.makeOffer ({
	            partnerSteamId: 'Steam id as string',
	            accessToken: 'access token as string',
	            itemsFromMe: [],
	            itemsFromThem: [ { appid: 730, contextid: 2, amount: 1, assetid: '469444882' } ],
	      }, function(err, response){
	            if (err) {
	              logger.error(err);
	            }
	      });
	    }
	});
}
```

But i get te same stack trace:

```
Error: There was an error sending your trade offer.  Please try again later. (26)
    at SteamTradeOffers.<anonymous> (/Users/diego/node_modules/steam-tradeoffers/index.js:427:18)
    at Request.self.callback (/Users/diego/node_modules/steam-tradeoffers/node_modules/request/request.js:368:22)
    at Request.emit (events.js:110:17)
    at Request.<anonymous> (/Users/diego/node_modules/steam-tradeoffers/node_modules/request/request.js:1219:14)
    at Request.emit (events.js:129:20)
    at IncomingMessage.<anonymous> (/Users/diego/node_modules/steam-tradeoffers/node_modules/request/request.js:1167:12)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

I have no idea what's happend.

**Comment by Commanche2** [comment]
As soon as I wrote the previous comment I realize what was my fault... :confounded: 
When I set the assetid on items structure, I put the classid (WROOONG!! If API says that you must put the assetid, you must put assetid and no other stuff!!) :cold_sweat:

```json
"332001318": {"id":"332001318","classid":"469444882","instanceid":"0","amount":"1","pos":191}
```
In this case, the assetid is **332001318** and isn't 469444882 as I supposed.

By the way, thank you for your time to try to solve my problem. I'll close this issue.
